src/services/ExplorerService.ts

```diff
@@ -76,39 +84,107 @@ export class ExplorerService implements IExplorerService {
     signer: any,
     transactionDetails: DelayedTransactionDetails
   ): Promise<void> {
-    if (this.featureScheduleTransaction) {
-      const polkadotApi = await this.polkadotApiService.getApi();
-
-      // Get the inner call using Polkadot API
-      const tx = polkadotApi.tx[transactionDetails.pallet][transactionDetails.extrinsic];
-      if (!tx) {
-        throw new Error(
-          `Invalid extrinsic: ${transactionDetails.pallet}.${transactionDetails.extrinsic}`
-        );
+    if (!this.tLockApi || !this.drandService) {
+      throw new Error('Timelock or Drand service not initialized');
+    }
+
+    const polkadotApi = await this.polkadotApiService.getApi();
+
+    // Get the inner call using Polkadot API
+    const tx = polkadotApi.tx[transactionDetails.pallet][transactionDetails.extrinsic];
+    if (!tx) {
+      throw new Error(
+        `Invalid extrinsic: ${transactionDetails.pallet}.${transactionDetails.extrinsic}`
+      );
+    }
+
+    // Parse parameters
+    const params = transactionDetails.params.map(param => {
+      if (param.value === 'true') return true;
+      if (param.value === 'false') return false;
+      if (!isNaN(param.value)) return Number(param.value);
+      return param.value;
+    });
+
+    // Create the inner call
+    const innerCall = tx(...params);
+
+    try {
+      // Get drand chain info for the public key
+      const chainInfo = await this.drandService.getChainInfo();
+      const drandPubKey = chainInfo.public_key;
+
+      // Serialize the inner call
+      const callData = innerCall.toU8a();
+
+      // Generate a random key for encryption (32 bytes)
+      const key = crypto.getRandomValues(new Uint8Array(32));
```

Comment:
The secret key should be computed using an hkdf, like so:

```
const computeSecretKey = async (seed) => {
    const hash = 'SHA-256'
    const length = 32
    const esk = await hkdf.compute(seed, hash, length, '')
    const key = Array.from(esk.key)
        .map((byte) => byte.toString(16).padStart(2, '0'))
        .join('')
    return key;
}
```

Reply:
change made!